juju/terraform-provider-juju
# Add data source for secrets

## Description
**This PR only has very basic acceptance tests right now. It will not be possible to do an end to end test until #454 is in as without this it is not possible for applications to make use of the secret.**

To access a secret in juju we are already adding in a new resource that allows you to create a new secret. But there may often be cases where the user does not wish to store the raw secret values in the terraform plan, and add them instead some other way. In this use case, a data source is needed.

A data source is added that allows a secret specified by name in a particular model to be referenced in the terraform plan.

## Type of change

- Add new data source

## Environment

- Juju controller version: 
3.3 and above

## QA steps

Manual QA steps should be done to test this PR.
- `juju bootstrap lxd`
- `juju add-model default`
- `juju add-secret test1 token=123123
- Run the terraform plan below and observe no errors.
```tf
terraform {
  required_providers {
    juju = {
      version = "~> 0.11.0"
      source  = "juju/juju"
    }
  }
}

provider "juju" {}

data "juju_model" "default" {
  name = "default"
}

data "juju_secret" "datasource-secretName" {
  name = "test1"
  model = data.juju_model.default.name
}
```
